core/src/main/java/bisq/core/xmr/connection/persistence/MoneroConnectionStore.java

```diff
@@ -0,0 +1,270 @@
+package bisq.core.xmr.connection.persistence;
+
+import bisq.core.api.CoreAccountService;
+import bisq.core.crypto.ScryptUtil;
+import bisq.core.util.Initializable;
+import bisq.core.xmr.connection.model.MoneroConnection;
+import bisq.core.xmr.connection.persistence.model.PersistableMoneroConnection;
+import bisq.core.xmr.connection.persistence.model.PersistableMoneroConnectionStore;
+
+import bisq.common.crypto.CryptoException;
+import bisq.common.crypto.Encryption;
+
+import com.google.protobuf.ByteString;
+
+import org.bitcoinj.crypto.KeyCrypterScrypt;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import javax.crypto.SecretKey;
+
+import java.security.SecureRandom;
+
+import java.net.URI;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.locks.Lock;
+import java.util.stream.Collectors;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Store for {@link MoneroConnection}s.
+ * <p>
+ * Passwords are encrypted when stored onto disk, using the account password.
+ * If a connection has no password, this is "hidden" by using some random value as fake password.
+ *
+ * @implNote The password encryption mechanism is handled as follows.
+ *  A random salt is generated and stored for each connection. If the connection has no password,
+ *  the salt is used as prefix and some random data is attached as fake password. If the connection has a password,
+ *  the salt is used as suffix to the actual password. When the password gets decrypted, it is checked whether the
+ *  salt is a prefix of the decrypted value. If it is a prefix, the connection has no password.
+ *  Otherwise, it is removed (from the end) and the remaining value is the actual password.
+ */
+@Singleton
+public class MoneroConnectionStore implements Initializable {
```

Comment:
Would like for this logic to be incorporated into `MoneroConnectionsManager` and/or `XmrConnectionList` rather than a separate store. I don't see any parallel for this structure in the existing services, and we're trying to maintain consistency throughout implementation.